packages/router/__tests__/router-test.ts

```diff
@@ -13544,4 +13555,60 @@ describe("a router", () => {
       });
     });
   });
+
+  describe("routes updates", () => {
+    it("should retain existing routes until revalidation completes", async () => {
+      let ref = {} as TMApiRef;
+      let t = initializeTmTest({ ref });
+      let ogRoutes = t.router.routes;
+      let A = await t.navigate("/foo");
+      await A.loaders.foo.resolve(null);
+      expect(t.router.state.loaderData).toMatchObject({
+        root: "ROOT",
+        foo: null,
+      });
+
+      let newRoutes = ref.enhanceRoutes([
+        {
+          path: "",
+          id: "root",
+          hasErrorBoundary: true,
+          loader: true,
+          children: [
+            {
+              path: "/",
+              id: "index",
+              loader: true,
+              action: true,
+            },
+            {
+              path: "/foo",
+              id: "foo",
+              loader: false,
+              action: true,
+            },
+          ],
+        },
+      ]);
+      t.router._internalSetRoutes(newRoutes);
+      t.router.revalidate();
```

Comment:
I think you can just lift the `let R = await t.revalidate();` up to here so there's only one revalidation happening?

Reply:
Thanks, was trying to figure out how to use this API.